src/module/chat-message/data.ts

```diff
@@ -26,6 +26,7 @@ type ChatMessageFlagsPF2e = foundry.data.ChatMessageFlags & {
         preformatted?: "flavor" | "content" | "both";
         isFromConsumable?: boolean;
         journalEntry?: DocumentUUID;
+        spellVariant?: { original: ItemUUID; id: string };
```

Comment:
Perhaps we should rename id to overrideId or override? We can later expand this to add the other types that involve a secondary choice.

Reply:
Actually, now I'm wondering what original is actually for here. Shouldn't origin.uuid be pointing to the original spell, not the variant?

Reply:
`original` is the UUID of the base spell (based on `SpellPF2e#original`) and `id` is the overlay id. The id field should actually be a string array now.
`originalSpell: ItemUUID` and `overlayIds: string[]` might be more descriptive.

Reply:
Is there a reason why the uuid of the variant spell, when cast, can't be the same uuid as the original spell? That way you won't need to do regex to handle that.

Reply:
I'm using the override overlay `id` as the document `_id` of the variant spell. That might no longer be necessary... I'll think about it.

Reply:
Ok, sheet editing breaks down when I omit the id from the override variants so I opted to override `SpellPF2e#uuid` and return the original uuid for variant spells. This way only the overlayIds need to be stored in the chat message flags.

Reply:
Ah, that's a shame. The only alternative I can think of is a keepId field that defaults to true that you can use when creating the variants for sheet usage by setting to false. But I'm fine with overriding uuid on the spell itself.

Reply:
Another possible idea might have been to override the sheet's id getter, if the problem with sheet editing is that you can't tell what's unique or not.

Reply:
Other than a keepId flag so that variants created for item sheets have a different id, I think this is the only other way. Honestly I'm fine with it. Overriding get uuid() is probably a solid call.